src/test/java/net/snowflake/client/jdbc/ConnectionIT.java

```diff
@@ -800,6 +803,36 @@ public void testStatementsAndResultSetsClosedByConnection() throws SQLException
     assertTrue(rs3.isClosed());
   }
 
+  @Test
+  public void testReadDateAfterSplittingResultSet() throws Exception {
+    Connection conn = getConnection();
+    try (Statement statement = conn.createStatement()) {
+      statement.execute("create or replace table table_with_date (int_c int, date_c date)");
+      statement.execute("insert into table_with_date values (1, '2015-10-25')");
+
+      try (ResultSet rs = statement.executeQuery("select * from table_with_date")) {
+        final SnowflakeResultSet resultSet = rs.unwrap(SnowflakeResultSet.class);
+        final List<SnowflakeResultSetSerializable> serializables =
+            resultSet.getResultSetSerializables(1 << 20);
+        final List<Date> dates =
+            serializables.stream()
+                .map(
+                    s -> {
+                      try {
+                        ResultSet srs = s.getResultSet();
+                        srs.next();
+                        return srs.getDate(2);
+                      } catch (Exception e) {
+                        throw new RuntimeException(e);
```

Comment:
Probably should throw some more appropriate exception (or even not catch in the first place)